toolkit/tools/scheduler/scheduler.go

```diff
@@ -187,6 +188,10 @@ func buildGraph(inputFile, outputFile string, agent buildagents.BuildAgent, work
 		return
 	}
 
+	if *hydratedBuild == "y" {
+		schedulerutils.ReplaceRunNodesWithPrebuiltNodes(pkgGraph, goalNode, packagesToBuild)
```

Comment:
I feel like this might actually be better placed before the initializeGraph() call, so that initializeGraph can take advantage of the new nodes to better prune.